Added Google Finance MCP Server

Badhri25 opened the pull request:
This pull request adds the Google Finance MCP Server.

- Implements atomic tools:
  - get_stock_price
  - get_historical_data
  - get_company_info
  - get_market_news
- All tools follow Klavis AI MCP guidelines.
- API key securely loaded via TWELVE_API_KEY environment variable.
- Includes comprehensive README.md and proof/ folder with screenshots.

Please see the proof/ folder for screenshots of successful runs.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Klavis-AI/klavis?pullRequest=223) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Klavis-AI/klavis?pullRequest=223) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Klavis-AI/klavis?pullRequest=223) it.</sub>

udaykakade25 commented:
Hey @Badhri25, You can start working on Google Finance MCP server in the directory:  `klavis/mcp_servers/google-finance`.

udaykakade25 commented:
@Badhri25 , for the folder structure, take reference from `affinity`. Stick to this format strictly.
.env .example, requirements.txt. dockerfile, etc, is missing.
Also, provide tests (screenshot or demo video) in the comments.

Badhri25 commented:
Here are the tests and proof of correctness for the Google Finance MCP server:

1. Server running and ready
2. /get_stock_price endpoint working
3. /get_historical_data endpoint working
4. /get_company_info endpoint working
5. /get_market_news endpoint working

Attaching screenshots below for each.

<img width="961" height="622" alt="server_running" src="https://github.com/user-attachments/assets/18bde30e-fd05-438a-b66d-7c6565e93741" />
<img width="907" height="361" alt="get_stock_price" src="https://github.com/user-attachments/assets/e1809780-2a6d-4411-9313-e5fe2e7dd774" />
<img width="906" height="324" alt="get_historical_data" src="https://github.com/user-attachments/assets/4b0f4de2-9012-4224-bb6f-b6974893d404" />
<img width="906" height="227" alt="get_company_info" src="https://github.com/user-attachments/assets/9ca8a047-c56f-4218-ac31-b733316cca33" />
<img width="907" height="232" alt="get_market_news" src="https://github.com/user-attachments/assets/f09a9dcf-4ea2-4c92-ae17-d305df404996" />

Badhri25 commented:
> @Badhri25 , for the folder structure, take reference from `affinity`. Stick to this format strictly. .env .example, requirements.txt. dockerfile, etc, is missing. Also, provide tests (screenshot or demo video) in the comments.

@udaykakade25 , I've updated the folder structure to match affinity and added `.env.example`, `requirements.txt`, and `Dockerfile`.  
Proof of correctness (screenshots) has also been added in the comments above.

Badhri25 commented:
@udaykakade25,  

Thanks for the feedback. Could you please share more details on what specific parts need refactoring or which standards I should align with?  
I'd be happy to make the necessary improvements and re-submit.

Badhri25 commented:
Hi @udaykakade25,  

Thank you for the detailed feedback. I’ll refactor the Google Finance MCP server to align with the guide and fix all the issues you mentioned (API key handling, __init__.py, missing tools, and Dockerfile).  

I’ll open a new PR once the changes are ready.  

Thanks!
